automate/scripts/etl_utils.py

```diff
@@ -0,0 +1,440 @@
+        
+        # Update version and MD5 hash
+        data["version"] = version
+        data["md5_hash"] = md5_hash
+        
+        # Write updated data back to the file
+        with open(file_path, 'w') as json_file:
+            json.dump(data, json_file, indent=4)
+        
+        print(f"JSON file '{file_path}' updated successfully.")
+    except Exception as e:
+        print(f"Error updating JSON file: {e}")
+
+def unzip_vocab(folder_name):
+    zip_local_file_path = f"../../{folder_name}/omop-vocab.zip"
+    unzip_at_path = "../../omop-vocab"
+    if not os.path.exists(unzip_at_path):
+            os.makedirs(unzip_at_path)
+            print(f"Created folder: {unzip_at_path}")
+    with zipfile.ZipFile(zip_local_file_path, 'r') as zip_ref:
+        zip_ref.extractall(os.path.dirname(unzip_at_path))
+        print(f"File 'omop-vocab.zip' unzipped successfully.")
+
+def calculate_retry_delay(attempt):
+    if attempt <= 5:
+        return min(2 ** (attempt - 1), 60)  # Exponential backoff up to 60 seconds
+    else:
+        return 60  # After 5 attempts, wait for 60 seconds
+
+
+# def extract_logs(input_file, output_file):
+#     try:
+#         with open(input_file, 'r') as f:
+#             lines = f.readlines()
+
+#         start_marker = "==== Summary ===="
+#         end_marker = "==== Job End ===="
+
+#         start_index = None
+#         end_index = None
+
+#         for i, line in enumerate(lines):
+#             if start_marker in line:
+#                 start_index = i
+#             elif end_marker in line:
+#                 end_index = i
+#                 break
+
+#         if start_index is not None and end_index is not None:
+#             extracted_lines = []
+
+#             for line in lines[start_index:end_index + 1]:
+#                 parts = line.split("FhirToOmopJobListener", 1)
+#                 if len(parts) > 1:
+#                     cleaned_line = parts[1].strip()
+#                     extracted_lines.append(cleaned_line + '\n')
+
+#             with open(output_file, 'w') as f:
+#                 f.writelines(extracted_lines)
+            
+#             print(f"Extracted logs and saved to '{output_file}'")
+#         else:
+#             print("Start or end marker not found.")
+
+#     except Exception as e:
+#         print(f"Error extracting logs: {e}")
+
+# def download_container_logs(yaml_path):
+#     try:
+#         # Run Docker Compose command in CMD
+#         with subprocess.Popen(f'docker-compose -f "{yaml_path}" logs ohdsi-germany > .././etl_report/full_report.txt', shell=True) as process:
+#             process.wait()  # Wait for the process to complete
+#         print("Docker logs downloaded ")
+#     except subprocess.CalledProcessError as e:
+#         print("Error running Docker Compose command:", e)
+#     except KeyboardInterrupt:
+#         print("Process interrupted by the user.")
+#         # Perform any necessary cleanup here
+#         raise
```

Comment:
Remove if not required.

Reply:
I think we can keep this code This code can be used to store container logs. if in future we want to store it any container logs